lib/core/src/lib/form/components/widgets/core/form-field.model.ts

```diff
@@ -289,6 +289,23 @@ export class FormFieldModel extends FormWidgetModel {
         return originalType === FormFieldTypes.DATETIME ? this.defaultDateTimeFormat : this.defaultDateFormat;
     }
 
+    private getId(id: string, parent?: RepeatableSectionModel): string {
+        return parent ? parent.uid : id;
+    }
+
+    private getRule(rule: FormFieldRule, parent?: RepeatableSectionModel): FormFieldRule {
+        return rule && parent
+            ? ({
+                  ...rule,
+                  ruleOn: this.getRepeatableSectionChildRuleOn(rule.ruleOn)
+              } as FormFieldRule)
+            : rule;
+    }
+
+    private getRepeatableSectionChildRuleOn(ruleOn: string) {
+        return ruleOn + ROW_ID_PREFIX + this.id.split(ROW_ID_PREFIX)[1];
+    }
+
```

Comment:
I think the responsibility about the rule can be handled by a new class that deal only with that. e.g. `form-field-rule-handler.ts`, or `form-field-rule-processor.ts`
This class is already a bit busy, shifting this logic to a dedicated class would keep responsibilities nicely separated